src/query_parser/v1/parser.y

```diff
@@ -225,15 +225,27 @@ union(A) ::= expr(B) OR expr(C) . [OR] {
         A->opts.fieldMask |= C->opts.fieldMask;
         QueryNode_SetFieldMask(A, A->opts.fieldMask);
     }
-    
 }
 
 union(A) ::= union(B) OR expr(C). [ORX] {
-    A = B;
-    if (C) {
+    int rv = one_not_null(B, C, (void**)&A);
+    if (rv == NODENN_BOTH_INVALID) {
+        A = NULL;
+    } else if (rv == NODENN_ONE_NULL) {
+        // Nothing- already assigned
+    } else {
+        if (B->type == QN_UNION && B->opts.fieldMask == RS_FIELDMASK_ALL) {
```

Comment:
Why is `RS_FIELDMASK_ALL` the condition for not having a new union?
Example?

Reply:
I see this was the condition from before as well.. If so not critical to the PR, just interested, if you're already touching it